docs/enhancements/multi-rhel-fips.md

```diff
@@ -0,0 +1,137 @@
+---
+title: fips-with-multiple-rhel-versions
+authors:
+  - "@carbonin"
+creation-date: 2024-05-07
+last-updated: 2024-05-08
+---
+
+# Support FIPS for installers built for different RHEL releases
+
+## Summary
+
+In order for an OpenShift cluster to be considered FIPS compliant the installer
+must be run on a system with FIPS compliant openssl libraries. This means using
+a dynamically linked `openshift-install` binary against the openssl libraries
+present on our container image. Today this is not a problem because all
+`openshift-install` binaries in use have been expecting to link to RHEL 8 based
+openssl libraries, but now OpenShift 4.16 will ship and installer that requires
+RHEL 9 libraries.
+
+This will require assisted-service to maintain a way to run the
+`openshift-install` binary in a compatible environment for multiple openssl
+versions.
+
+## Motivation
+
+FIPS compliance is important for our customers and assisted-service should be
+able to install FIPS compliant clusters.
+
+### Goals
+
+- Allow for a single installation of assisted-service to install FIPS-compliant
+  clusters using `openshift-install` binaries built against RHEL 8 or RHEL 9
+
+- Allow for FIPS compliant clusters to be installed from the SaaS offering or
+  the on-prem offering
+
+### Non-Goals
+
+- Changing cluster install interfaces to accommodate new FIPS requirements
+  should be avoided
+
+- Dynamically determining a given release's RHEL version. The RHEL version will
+  be a part of configuration tied to the minor release
```

Comment:
Out of curiosity, how will this look? Will it be an index we maintain like the default_os/release images files?

Reply:
I was thinking that, but now I'm rethinking it.

Maybe it would be better to just switch on if the version is 4.16 or later.
@mhrivnak mentioned OKD in this context. What do the cluster versions look like when OKD is in use?

Reply:
Image list trimmed:

```
oc adm release info quay.io/openshift/okd@sha256:46b462be1e4c15ce5ab5fba97e713e8824bbb9f614ac5abe1be41fda916920cc
Name:           4.15.0-0.okd-2024-03-10-010116
Digest:         sha256:46b462be1e4c15ce5ab5fba97e713e8824bbb9f614ac5abe1be41fda916920cc
Created:        2024-03-10T06:53:44Z
OS/Arch:        linux/amd64
Manifests:      707
Metadata files: 1

Pull From: quay.io/openshift/okd@sha256:46b462be1e4c15ce5ab5fba97e713e8824bbb9f614ac5abe1be41fda916920cc

Release Metadata:
  Version:  4.15.0-0.okd-2024-03-10-010116
  Upgrades: <none>
  Metadata:

Component Versions:
  kubernetes 1.28.7        
  machine-os 39.20240210.3 Fedora CoreOS
```

Reply:
So we'd still get `4.15.0-0.okd-2024-03-10-010116` in this case.

I mean I suppose if no one is doing FIPS with OKD then we should be able to apply the same rule and it will work out. (4.16 OKD would go to the el9 runner)

Reply:
why not always use fips compliant installer?

Reply:
We will, but which FIPS compliant installer we use (rhel8 or rhel9) will be determined by the release version.